### PR TITLE
feat: Add support for increasing partitions to the kafka shim

### PR DESCRIPTION
**Pull request by palmere-google**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-pubsublite-kafka/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteProducerTest.java [hunk 1]
```diff
@@ -23,30 +23,29 @@
...
+import com.google.api.core.ApiFutures;
 import com.google.api.core.SettableApiFuture;
 import com.google.api.gax.rpc.StatusCode.Code;
-import com.google.cloud.pubsublite.Message;
-import com.google.cloud.pubsublite.Offset;
-import com.google.cloud.pubsublite.Partition;
-import com.google.cloud.pubsublite.PublishMetadata;
-import com.google.cloud.pubsublite.TopicName;
-import com.google.cloud.pubsublite.TopicPath;
+import com.google.cloud.pubsublite.*;
```
Comment (dpcollins-google) [inline src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteProducerTest.java:29]: no *

File: src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java [hunk 2]
```diff
@@ -30,13 +30,7 @@
...
 import com.google.api.core.ApiFutures;
 import com.google.api.core.SettableApiFuture;
-import com.google.cloud.pubsublite.CloudZone;
-import com.google.cloud.pubsublite.Offset;
-import com.google.cloud.pubsublite.Partition;
-import com.google.cloud.pubsublite.ProjectNumber;
-import com.google.cloud.pubsublite.SubscriptionPath;
-import com.google.cloud.pubsublite.TopicName;
-import com.google.cloud.pubsublite.TopicPath;
+import com.google.cloud.pubsublite.*;
```
Comment (dpcollins-google) [inline src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java:33]: No *

File: src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java [hunk 3]
```diff
@@ -18,17 +18,24 @@
...
 import com.google.common.collect.ImmutableList;
+import java.time.Duration;
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 import org.apache.kafka.common.PartitionInfo;
 
 /** Shared behavior for producer and consumer. */
 final class SharedBehavior {
-  private SharedBehavior() {}
+  AdminClient client;
```
Comment (dpcollins-google) [inline src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java:32]: This needs to have close() called on it. If you want to do this, you need to make SharedBehavior implement AutoCloseable.

**Review by dpcollins-google — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (palmere-google) [reply src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java:32]: Done

File: src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteProducerTest.java (hunk 1 above)
Comment (palmere-google) [reply src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteProducerTest.java:29]: Sorry, my editor does this. I'll try to make it stop. (Fixed)

File: src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java (hunk 2 above)
Comment (palmere-google) [reply src/test/java/com/google/cloud/pubsublite/kafka/PubsubLiteConsumerTest.java:33]: Done

File: src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java (hunk 3 above)
Comment (palmere-google) [reply src/main/java/com/google/cloud/pubsublite/kafka/SharedBehavior.java]: Ah, right. Done. I forgot about Closeable, thanks!

**Review by dpcollins-google — APPROVED** [review]

**Pull request merged**